### PR TITLE
Magic Mania 1.85: RPG loot event

### PR DESCRIPTION
**Pull request by Incoming5643**
Adds the RPG loot wizard event (summon events only) at an uncommon (3) weight. One time only run event.

All items in the world at the time of event gain a small bonus/penalty to force and throwforce, and a corresponding silly name. Less lag inducing then you probably think.

Taking all suggestions for prefixes and suffixes. Prefixes should imply a positive or negative quality, but suffixes shouldn't. Feel free to get a little wacky.

### CODE REVIEW

**Comment by Miauw62** [comment]
yes

**Comment by TZK13** [comment]
What about armour values or is that too much.

**Comment by Incoming5643** [comment]
We could but I'm intentionally not going too complicated at the start here. Something to add later maybe.

**Comment by TZK13** [comment]
Another idea for the future could be colouring items to correspond with rarity.

File: code/modules/events/wizard/rpgloot.dm [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		var/quality = rand(-5,5)
+		if(quality > 0)
+			I.name = "[pick(prefixespositive)] [I.name] of [pick(suffixes)] +[quality]"
+		else if(quality < 0)
+			I.name = "[pick(prefixesnegative)] [I.name] of [pick(suffixes)] [quality]"
+		else
+			I.name = "[I.name] of [pick(suffixes)]"
+
+		I.force 		+= quality
+		I.force			= max(0,quality)
```
Comment (Mloc) [inline code/modules/events/wizard/rpgloot.dm]: should be just `I.force = max(0, I.force +quality)`, currently it'll just overwrite to `quality`.

**Comment by Miauw62** [comment]
it should make some items "cursed" and give them NODROP.

**Comment by Alek2ander** [comment]
The enchantment looks better in front of the name. Also, yeah, blessed/cursed. Maybe use this system for the "cursed items" event as well so everyone gets a different katana.

**Comment by WJohn** [comment]
I'm a bit curious about nodrop, what happens if someone gets turned into a monkey while holding an item like that? Does the item suddenly drop or does it stick to their monkey hands anyway?

**Comment by Miauw62** [comment]
That depends on whether I made Monkeyize() use the force parameter or not.
I don't think I did, so it's just deleted, probably.
Op 2-feb.-2015 12:55 schreef "WJohn" <notifications@github.com>:

**Comment by WJohn** [comment]
The item just gets deleted? Someone should test the nuke disk with that and see if it respawns properly.

**Comment by Miauw62** [comment]
nuke disk won't ever be NODROP anyway.
it gets deleted because the old mob is deleted.

**Comment by MrPerson** [comment]
I'm not a fan of the suffixes that imply stat gains, like of constitution. People will (correctly) assume that it will actually increase your constitution. And I really, really don't want to encourage people to run around with a special pair of wirecutters that gives them poison resistance or shit like that, so I'm against actually giving them those kinds of traits. Thus I feel that the suffixes that imply bonuses like that should just be removed.

This should only be allowed to run once or else you'll get shit like "blighted greater wrench of corgi-slaying of many colors +3 -2" which is really stupid.

**Comment by Iamgoofball** [comment]
You should color the names too

**Comment by Incoming5643** [comment]
Currently it's only allowed to run once.

When I was using classic D20 stats for suffixes I was thinking more of stats that work off them rather than raise them, like a charisma based rapier for bards. The joke of course being that since SS13 doesn't have a physical stats system, it doesn't actually proffer any sort of benefit.

**Comment by Tenebrosity** [comment]
I like the idea a lot, but if it only increases force by a maximum of 5, it's fairly useless functionally

**Comment by Incoming5643** [comment]
Welcome to common/uncomming summon events events

**Comment by Tenebrosity** [comment]
Well, if the event is uncommon and can only happen once, surely you can risk setting the maximum buff amount higher than 5 points? It would pretty rare that a weapon would get in the upper quartile of damage modification. I'm just saying this because the event serves to buff or nerf melee weapons, but it's by such an insignificant amount that it seems pointless. Just my two cents.

**Comment by Kromgar** [comment]
+5 Toolbox of Smiting go

**Comment by TZK13** [comment]
>No X of robusting.

Are you serious mate.

**Comment by Incoming5643** [comment]
I said I was taking suggestions man, throw them at me

**Comment by Jordie0608** [comment]
Changelog.

**Comment by Incoming5643** [comment]
So...

File: code/modules/events/wizard/rpgloot.dm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+			I.name = "[pick(prefixespositive)] [I.name] of [pick(suffixes)] +[quality]"
+		else if(quality < 0)
+			I.name = "[pick(prefixesnegative)] [I.name] of [pick(suffixes)] [quality]"
+		else
+			I.name = "[I.name] of [pick(suffixes)]"
+
+		I.force 		+= quality
+		I.force			= max(0,I.force)
+		I.throwforce 	+= quality
+		I.throwforce	= max(0,I.throwforce)
```
Comment (phil235) [inline code/modules/events/wizard/rpgloot.dm]: maybe add `ìf(I.force)` so that only items that can already be used as weapon gets changed.

Comment (Incoming5643) [inline code/modules/events/wizard/rpgloot.dm]: items that aren't weapons (literally not /weapons/) don't attack when you use them, so while they can have an force, it's irrelevant. There's no need to protect the field like that.

Comment (phil235) [inline code/modules/events/wizard/rpgloot.dm]: I hit and knocked down a mob with an obj/item/clothing . It's probably better to restrict the effect to obj/item/weapon then.

**Pull request merged**